tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -93,6 +94,7 @@ void ChangeBeginAndEndEntries(const RNode &node, Long64_t begin, Long64_t end);
 void ChangeSpec(const ROOT::RDF::RNode &node, ROOT::RDF::Experimental::RDatasetSpec &&spec);
 void TriggerRun(ROOT::RDF::RNode node);
 std::string GetDataSourceLabel(const ROOT::RDF::RNode &node);
+void SetTTreeLifeline(ROOT::RDF::RNode &node, const std::any &lifeline);
```

Comment:
```suggestion
void SetTTreeLifeline(ROOT::RDF::RNode &node, std::any lifeline);
```
...and then `std::move()` the `lifeline` inside the function to elide a copy. Also makes the signature more compact and highlights the fact that the wrapped type should be cheaply copyable.

Also, maybe consider a more general name like `SetSharedResource`? I'm thinking that we will also re-use this interface to manage other resources from the Python side, like NumPy arrays and so on

Reply:
> Also, maybe consider a more general name like SetSharedResource? I

This function is specifically meant for setting the value of the data member of the RLoopManager. Before making it more generic, I would like to see how we would use the same pattern for another class (e.g. the RVecDS). I would say that's for another PR